aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -13,80 +13,132 @@ namespace native {
 
 constexpr size_t dim_bitset_size = 64;
 
-template <typename scalar_t>
-void inline flip_cpu_kernel(
-  const int64_t total_dims,
-  const std::vector<int64_t>& stride_contiguous_v,
-  const std::bitset<dim_bitset_size>& flip_dims_b,
-  const Tensor& in_tensor,
-  Tensor& out_tensor
-){
-  const int64_t numel = in_tensor.numel();
-  const scalar_t* in_tensor_d = in_tensor.data_ptr<scalar_t>();
-  scalar_t* out_tensor_d = out_tensor.data_ptr<scalar_t>();
-  auto sizes_v = in_tensor.sizes().vec();
-  auto strides_v = in_tensor.strides().vec();
-
-  at::parallel_for(0, numel, 1000, [&](int64_t start, int64_t end) {
-    for (auto i = start; i < end; i++) {
-      int64_t cur_indices = i;
-      int64_t rem = 0;
-      int64_t dst_offset = 0;
-
-      for (int64_t d = 0; d < total_dims; d++) {
-        int64_t temp = cur_indices;
-        cur_indices = cur_indices / stride_contiguous_v[d];
-        rem = temp - cur_indices * stride_contiguous_v[d];
-        dst_offset += flip_dims_b[d] ? (sizes_v[d] - 1 - cur_indices) * strides_v[d] : cur_indices * strides_v[d];
-        cur_indices = rem;
-      }
-      out_tensor_d[i] = in_tensor_d[dst_offset];
+struct Indexer {
+  Indexer(int64_t num_indexers, char** indexers, const int64_t* indexer_strides)
+    : num_indexers(num_indexers)
+    , indexers(indexers)
+    , indexer_strides(indexer_strides) {
+
+  }
+
+  int64_t num_indexers;
+  char** indexers;
+  const int64_t* indexer_strides;
+
+  int64_t get(int64_t idx) {
+    int64_t offset = *(int64_t*)&indexers[0][idx * indexer_strides[0]];
+    for (int j = 1; j < num_indexers; j++) {
+      offset += *(int64_t*)&indexers[j][idx * indexer_strides[j]];
```

Comment:
For the future: we could look into specializing this when `num_indexers=1`. It could bring additional performance improvements